lib/graphql/metrics/instrumentation.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module GraphQL
+  module Metrics
+    class Instrumentation
+      def before_query(query)
+        unless query_present_and_valid?(query)
+          query.context[GraphQL::Metrics::SKIP_GRAPHQL_METRICS_ANALYSIS] = true
```

Comment:
Maybe just inline the two guard clauses and `return` early? I found it slightly confusing that you set the skip flag here instead of just directly returning